libfreerdp/codec/rfx_neon.c

```diff
@@ -225,6 +225,296 @@ static void rfx_dwt_2d_decode_NEON(INT16* buffer, INT16* dwt_buffer)
 	rfx_dwt_2d_decode_block_NEON(buffer, dwt_buffer, 32);
 }
 
+static INLINE void __attribute__((__gnu_inline__, __always_inline__, __artificial__))
+rfx_idwt_extrapolate_horiz_neon(const INT16* restrict pLowBand, size_t nLowStep,
+                                const INT16* restrict pHighBand, size_t nHighStep,
+                                INT16* restrict pDstBand, size_t nDstStep, size_t nLowCount,
+                                size_t nHighCount, size_t nDstCount)
+{
+	int y, n;
+	INT16* l_ptr = pLowBand;
+	INT16* h_ptr = pHighBand;
+	INT16* dst_ptr = pDstBand;
+	size_t batchSize = (nLowCount + nHighCount) >> 1;
+
+	for (y = 0; y < nDstCount; y++)
```

Comment:
we´re using modern `C` now, you can declare the variable in the loop like `for (size_t y=0; y<nDstCount; y++)`
also ensure the type matches (you have `int y` and `size_t nDstCount` here)